ci/scripts/python_build.sh

```diff
@@ -37,7 +37,7 @@ export PYARROW_WITH_FLIGHT=${ARROW_FLIGHT:-OFF}
 export PYARROW_WITH_PLASMA=${ARROW_PLASMA:-OFF}
 export PYARROW_WITH_GANDIVA=${ARROW_GANDIVA:-OFF}
 export PYARROW_WITH_PARQUET=${ARROW_PARQUET:-OFF}
-export PYARROW_WITH_DATASET=${ARROW_DATASET:-OFF}
```

Comment:
@jorisvandenbossche nice catch, but ARROW_DATASET should be set from the docker images, like  https://github.com/apache/arrow/blob/master/ci/docker/conda-cpp.dockerfile#L65

Reply:
Actually in each `*-cpp.dockerfile`